tests/util/caches/test_descriptors.py

```diff
@@ -703,6 +704,48 @@ async def list_fn(self, args1, arg2):
             obj.mock.assert_called_once_with((40,), 2)
             self.assertEqual(r, {10: "fish", 40: "gravy"})
 
+    def test_concurrent_lookups(self):
+        """All concurrent lookups should get the same result"""
+
+        class Cls:
+            def __init__(self):
+                self.mock = mock.Mock()
+
+            @descriptors.cached()
+            def fn(self, arg1):
+                pass
+
+            @descriptors.cachedList("fn", "args1")
+            def list_fn(self, args1) -> Deferred[dict]:
```

Comment:
Oh, CI is failing because the old version of twisted doesn't have generic `Deferred`. I think you want `"Deferred[dict]"` here.

```
2021-11-04T12:47:21.2343360Z ===============================================================================
2021-11-04T12:47:21.2344758Z [ERROR]
2021-11-04T12:47:21.2345162Z Traceback (most recent call last):
2021-11-04T12:47:21.2345884Z   File "/github/workspace/tests/util/caches/test_descriptors.py", line 710, in test_concurrent_lookups
2021-11-04T12:47:21.2346531Z     class Cls:
2021-11-04T12:47:21.2348968Z   File "/github/workspace/tests/util/caches/test_descriptors.py", line 719, in Cls
2021-11-04T12:47:21.2350189Z     def list_fn(self, args1) -> Deferred[dict]:
2021-11-04T12:47:21.2351885Z builtins.TypeError: 'type' object is not subscriptable
2021-11-04T12:47:21.2352322Z 
2021-11-04T12:47:21.2353364Z tests.util.caches.test_descriptors.CachedListDescriptorTestCase.test_concurrent_lookups
2021-11-04T12:47:21.2355276Z -------------------------------------------------------------------------------
```

Reply:
yes I did. Thanks.